elfpy/utils/apeworx_integrations.py

```diff
@@ -167,92 +166,121 @@ def get_on_chain_trade_info(hyperdrive_contract: ContractInstance) -> OnChainTra
         Contract pointing to the base currency (e.g. ERC20)
+    index : int
+        Index of the wallet among ALL agents.
 
     Returns
     -------
     Wallet
         Wallet with Short, Long, and LP positions.
     """
     # TODO: remove restriction forcing Wallet index to be an int (issue #415)
-    wallet = Wallet(
-        address=index,
-        balance=types.Quantity(amount=FixedPoint(base_contract.balanceOf(address_)), unit=types.TokenType.BASE),
-    )
+    if add_to_existing_wallet is None:
+        wallet = Wallet(
+            address=index,
+            balance=types.Quantity(
+                amount=FixedPoint(scaled_value=base_contract.balanceOf(address)), unit=types.TokenType.BASE
+            ),
+        )
+    else:
+        wallet = add_to_existing_wallet
     for position_id in info.unique_ids:  # loop across all unique positions
-        trades_in_position = ((info.trades["from"] == address_) | (info.trades["to"] == address_)) & (
+        trades_in_position = ((info.trades["from"] == address) | (info.trades["to"] == address)) & (
             info.trades["id"] == position_id
         )
-        log_and_show("found %s trades for %s in position %s", sum(trades_in_position), address_[:8], position_id)
-        balance = (
-            info.trades.loc[(trades_in_position) & (info.trades["to"] == address_), "value"].sum()
-            - info.trades.loc[(trades_in_position) & (info.trades["from"] == address_), "value"].sum()
+        log_and_show("found %s trades for %s in position %s", sum(trades_in_position), address[:8], position_id)
+        positive_balance = int(info.trades.loc[(trades_in_position) & (info.trades["to"] == address), "value"].sum())
+        negative_balance = int(info.trades.loc[(trades_in_position) & (info.trades["from"] == address), "value"].sum())
+        balance = positive_balance - negative_balance
+        logging.debug(
+            "balance %s = positive_balance %s - negative_balance %s", balance, positive_balance, negative_balance
         )
         asset_prefix, maturity = hyperdrive_assets.decode_asset_id(position_id)
         asset_type = AssetIdPrefix(asset_prefix).name
         mint_time = maturity - elfpy.SECONDS_IN_YEAR
         log_and_show(f" => {asset_type}({asset_prefix}) maturity={maturity} mint_time={mint_time}")
+
+        on_chain_balance = 0
         # verify our calculation against the onchain balance
-        on_chain_balance = hyperdrive_contract.balanceOf(position_id, address_)
-        if abs(balance - on_chain_balance) > elfpy.MAXIMUM_BALANCE_MISMATCH_IN_WEI:
-            raise ValueError(
-                f"events {balance=} and {on_chain_balance=} disagree by "
-                f"more than {elfpy.MAXIMUM_BALANCE_MISMATCH_IN_WEI} wei for {address_}"
-            )
-        log_and_show(f" => calculated balance = on_chain = {fmt(balance)}")
+        if add_to_existing_wallet is None:  # sourcery skip: merge-nested-ifs
```

Comment:
I don't want to see sourcery comments in the code.  skip it in your local config